.github/workflows/run_tests_incremental.yml

```diff
@@ -96,34 +91,43 @@ jobs:
           # See https://github.com/shadow/shadow/issues/2166
           ref: ${{ github.event.pull_request.head.sha }}
 
-      - name: Get month
-        id: get-month
+      - name: Test
         run: |
-          echo "::set-output name=month::$(/bin/date -u "+%Y%m")"
+          FLAGS=()
+          FLAGS+=("-c" "${{ matrix.container }}")
+          FLAGS+=("-C" "${{ matrix.cc }}")
+          FLAGS+=("-b" "${{ matrix.buildtype }}")
 
-      - name: Install dependencies
-        run: |
-          . ci/container_scripts/install_deps.sh
-          . ci/container_scripts/install_extra_deps.sh
-          echo "$HOME/.cargo/bin" >> $GITHUB_PATH
+          # Get the generated tag for the image
+          IMAGETAG=`ci/run.sh -t ${FLAGS[@]}`
 
-      - name: Restore cargo registry cache
-        uses: actions/cache@v3
-        with:
-          path: |
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-          # invalidate the cache once per month
-          key: cargo-registry-${{ steps.get-month.outputs.month }}
-          restore-keys: |
-            cargo-registry-
-
-      - name: Build
-        run: . ci/container_scripts/build_and_install.sh
+          if [ "${{ github.event.inputs.build}}" == "true" ]
+          then
+            # Build the image
+            FLAGS+=("-i")
```

Comment:
In the workflow dispatch API, booleans are actually strings and you could call the API with a value containing a `"` to break out of this bash string and run arbitrary code. Since we're the only ones who can run it, this is probably fine, but I think it's generally better to use the environment variable format for inputs to prevent that (ex: `env: INPUT_BUILD: ${{ github.event.inputs.build}}`).

Reply:
oof, nice catch